### PR TITLE
encryption: fix encryption bootstrap logic

### PR DESCRIPTION
**Pull request by sttts**
- key controller starts with the first key when the mode changes
- state controller follows key controller only
- when config or keys exists, resetting mode to "" has no effect,
  other than being understood as identity, but leaves encryption
  mechanism active.

### CODE REVIEW

File: pkg/operator/encryption/state.go [hunk 1]
```diff
@@ -26,13 +26,18 @@ func getCurrentEncryptionConfig(secrets corev1client.SecretInterface, revision s
...
 		if errors.IsNotFound(err) {
-			return &apiserverconfigv1.EncryptionConfiguration{}, nil
+			return nil, nil
 		}
 		return nil, err
 	}
 
+	data, ok := encryptionConfigSecret.Data[encryptionConfSecret]
+	if !ok {
+		return nil, nil
```
Comment (p0lyn0mial) [inline pkg/operator/encryption/state.go]: does it mean that the current encryption config is broken?
shouldn't we report an error in this case?

Comment (sttts) [reply pkg/operator/encryption/state.go]: We just behave like for a missing config. We can recover.

**Pull request merged**